### ISSUE TITLE
For unbundled builds, be able to configure reference to be inlined.

### ISSUE THREAD

**Issue by bahrus** [Polymer/polymer-cli#783]
### Description
In the scheme of things, this is probably a low priority request (at best).  Just wanted to get it off my chest, in case this makes sense to anyone else.

There is no established protocol for editing / handing typescript  script tags within an html document.  Hence those individuals who want to utilize typescript are pretty much forced to separate the code from the HTML file.  Maybe that's a good thing anyway, and some none typescripters would also prefer to keep things separate during development.

But while debugging in chrome, It would be kind of nice to see the class definition combined in the same "file" as the html template.  That way if you jump to the function definition, you can see the template right there.    Likewise, even for unbundled, HTTP/2 settings, there _might_ be a small performance benefit from inlining the script file into the html file.  The benefit would only be there if the script file is tightly coupled to a single html file, which will typically be the case for Polymer components.

So my proposal is to support at least one of these options:

- Provide a build setting in polymer.json, for unbundled builds, to always inline script files if they are only referenced by a single html file (with the same name?).  Make polymer serve apply the same logic.  Or

- Support some sort of custom attribute on the script tag that indicates the developer wants the script to be embedded in the html during builds (and when serving via polymer serve).  Or

-  Just aggressively do the inlinining without any config setting, for both the unbundled build, and the polymer serve, if there is no question that performance is improved in all cases by doing this, and if the polymer analyzer is 100% confident the script file is tightly coupled to the html file.

**Comment by justinfagnani** [comment]
This issue was moved to Polymer/polymer-bundler#533

**Comment by usergenic** [comment]
Re-opening this issue and closing Polymer/polymer-bundler#533 because this is either a build or cli concern and not a bundling concern AFAICT

**Comment by usergenic** [comment]
We may need to wait on this until Polymer Analyzer supports Typescript, which we are aiming for.
